packages/tpetra/core/test/ImportExport2/ImportExport2_CrsSortingUtils.cpp

```diff
@@ -0,0 +1,415 @@
+
+    //
+    // Sort the GIDs and associated values
+    //
+
+    // Create array views of data
+    auto rowptr_av = Teuchos::ArrayView<size_t>(rowptr2);
+    auto colind_rand_av = Teuchos::ArrayView<GO>(colind_rand);
+    auto vals_rand_av = Teuchos::ArrayView<Scalar>(vals_rand);
+
+    // Do the sort
+    Tpetra::Import_Util::sortAndMergeCrsEntries<Scalar, GO>(rowptr_av, colind_rand_av, vals_rand_av);
+
+    // At this point, the row pointers and column indices should be the same as
+    // the version above due to the merge/shrink
+    size_type new_num_entries = rowptr2[rowptr2.size()-1];
+    TEST_EQUALITY(colind.size(), new_num_entries);
+
+    TEST_COMPARE_ARRAYS(rowptr, rowptr2);
+
+    colind_rand.resize(new_num_entries);
+    TEST_COMPARE_ARRAYS(colind, colind_rand);
+
+    // At this point, the values should be twice the vals above due to the
+    // merge/shrink
+    vals2.resize(new_num_entries);
+    vals_rand.resize(new_num_entries);
+    for (size_type i=0; i<new_num_entries; i++) vals2[i] = 2.*vals[i];
+    TEST_COMPARE_FLOATING_ARRAYS(vals2, vals_rand, 1.e-12);
+
+    //
+    // Sort the GIDs w/o values
+    //
+
+    // Create array views of data
+    auto rowptr_copy_av = Teuchos::ArrayView<size_t>(rowptr2_copy);
+    auto colind_rand_copy_av = Teuchos::ArrayView<GO>(colind_rand_copy);
+
+    // Do the sort
+    Tpetra::Import_Util::sortAndMergeCrsEntries<GO>(rowptr_copy_av, colind_rand_copy_av);
+
+    // At this point, the row pointers and column indices should be the same as
+    // the version above due to the merge/shrink
+    new_num_entries = rowptr2_copy[rowptr2_copy.size()-1];
+    TEST_EQUALITY(colind.size(), new_num_entries);
+
+    TEST_COMPARE_ARRAYS(rowptr, rowptr2_copy);
+
+    colind_rand_copy.resize(new_num_entries);
+    TEST_COMPARE_ARRAYS(colind, colind_rand_copy);
+
+  }
+
+}
+
+TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( Import_Util, SortCrsEntriesKokkos, Scalar, LO, GO, NT)
+{
+
+  typedef typename Tpetra::CrsMatrix<Scalar,LO,GO,NT>::local_matrix_type local_matrix_type;
+  typedef typename local_matrix_type::StaticCrsGraphType graph_type;
+  typedef typename graph_type::row_map_type::non_const_type rowptr_view_type;
+  typedef typename graph_type::entries_type::non_const_type colind_view_type;
+  typedef typename local_matrix_type::values_type::non_const_type vals_view_type;
+
+  typedef typename std::vector<typename rowptr_view_type::value_type> rowptr_type;
+  typedef typename std::vector<typename colind_view_type::value_type> colind_type;
+  typedef typename std::vector<typename vals_view_type::value_type> vals_type;
+  typedef typename colind_type::size_type size_type;
+
+  // Map is not actually used, but is needed to instantiate Kokkos
+  auto comm = Tpetra::DefaultPlatform::getDefaultPlatform().getComm();
+  const Tpetra::global_size_t INVALID =
+    Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid();
+  auto dummy_map = Tpetra::Map<LO,GO,NT>(INVALID, 1, 0, comm);
```

Comment:
This is fine for now, but I would recommend the "different unit test main()" approach in the future.